includes/payment-methods/class-wc-stripe-upe-payment-method-cc.php

```diff
@@ -41,8 +41,9 @@ public function __construct() {
 	 * @return string
 	 */
 	public function get_title( $payment_details = false ) {
-		if ( $payment_details && isset( $payment_details->card->wallet->type ) ) {
-			return $this->get_card_wallet_type_title( $payment_details->card->wallet->type );
+		$wallet_type = WC_Stripe_Payment_Methods::AMAZON_PAY === ( $payment_details->type ?? null ) ? WC_Stripe_Payment_Methods::AMAZON_PAY : ( $payment_details->card->wallet->type ?? null );
```

Comment:
Maybe I should replace this with another IF. A bit confusing to read.

Reply:
I agree an if-else clause might be easier to read here. 👍